core/Policy/PolicyManager.php

```diff
@@ -103,12 +103,47 @@ public static function isPolicyActive(string $policyClass, ?int $idSite = null):
      * @throws \Exception when $policyClass is not a valid policy
      */
     public static function getAllUnknownSettings(string $policyClass): array
+    {
+        self::checkPolicyIsValid($policyClass);
+
+        return $policyClass::getUnknownSettings();
+    }
+
+    /**
+     * @param class-string<CompliancePolicy> $policyClass
+     * @throws \Exception when $policyClass is not a valid policy
+     */
+    private static function checkPolicyIsValid(string $policyClass): void
     {
         if (!is_a($policyClass, CompliancePolicy::class, true)) {
             throw new Exception('Invalid compliance policy.');
         }
+    }
 
-        return $policyClass::getUnknownSettings();
+    /**
+     * @param class-string<CompliancePolicy> $policyClass
+     * @throws \Exception when $policyClass is not a valid policy
+     */
+    public static function setPolicyActiveStatus(string $policyClass, bool $isActive, ?int $idSite = null): void
+    {
+        self::checkPolicyIsValid($policyClass);
+
+        $policyClass::setActiveStatus($idSite, $isActive);
+        Cache::deleteTrackerCache();
+    }
+
+    public static function storePolicySettingValuesInTrackerCache(array &$cacheContent, int $idSite): array
```

Comment:
Could/should we call this from somewhere else other than the privacy manager via a hook to the event? Could there be compliance policies active without the privacy manager? If so, the cache wouldn't be populated. Maybe a hook in CorePluginsAdmin or CoreHome? Haven't looked for a better place in code, just doesn't feel 100% right being hooked only in the PrivacyManager plugin.